src/engine/cachingreader.cpp

```diff
@@ -201,15 +201,42 @@ CachingReaderChunkForOwner* CachingReader::lookupChunkAndFreshen(SINT chunkIndex
 }
 
 void CachingReader::newTrack(TrackPointer pTrack) {
+    // Feed the track to the worker as soon as possible
+    // to get ready while the reader switches its internal
+    // state. There are no race conditions, because the
+    // reader polls the worker.
     m_worker.newTrack(pTrack);
     m_worker.workReady();
+    // Don't accept any new read requests until the current
+    // track has been unloaded and the new track has been
+    // loaded.
+    m_state = State::TrackLoading;
+    // Free all chunks with sample data from the current track.
+    freeAllChunks();
 }
 
 void CachingReader::process() {
     ReaderStatusUpdate update;
-    while (m_readerStatusFIFO.read(&update, 1) == 1) {
+    while (m_stateFIFO.read(&update, 1) == 1) {
+        DEBUG_ASSERT(m_state != State::Idle);
```

Comment:
- After TRACK_LOADED the state becomes State::TrackLoaded
- After TRACK_UNLOADED the state becomes State::Idle and no new updates are expected.

While the reader is *idle* no update messages are expected!

Reply:
The debug assertion is there for a reason. Idle means don't bother me with update messages or something is seriously wrong.

Reply:
This might be true after your recent changes in the other PR, but this assumption is here wrong. 
Try it out. I am still wrapping my head around you new code but I want first understand the original issue.

Was this your assertion as well?